ExtractUserDataActivity/handler.ts

```diff
@@ -432,6 +464,71 @@ export const createExtractUserDataActivityHandler = (
         }
       );
 
+  /**
+   * Creates a bundle with all user data and save it to a blob on a remote storage
+   * @param data all extracted user data
+   * @param password a password for bundle encryption
+   *
+   * @returns either a failure or an object with the name of the blob and the password
+   */
+  const saveDataToBlob = (
+    data: AllUserData,
+    password: StrongPassword
+  ): TaskEither<ActivityResultArchiveGenerationFailure, ArchiveInfo> =>
+    taskify(
+      (
+        cb: (
+          e: ActivityResultArchiveGenerationFailure | null,
+          r?: ArchiveInfo
+        ) => void
+      ) => {
+        const blobName = `${
+          data.profile.fiscalCode
+        }-${Date.now()}.zip` as NonEmptyString;
+        const fileName = `${data.profile.fiscalCode}.json` as NonEmptyString;
+
+        const readableZipStream = createCompressedStream(
+          {
+            [fileName]: data
+          },
+          password
+        );
+
+        const writableBlobStream = blobService.createWriteStreamToBlockBlob(
+          userDataContainerName,
+          blobName,
+          (err, _) => {
```

Comment:
it's not clear why do we need the callback here ? cannot we just call 

` const writableBlobStream = blobService.createWriteStreamToBlockBlob(userDataContainerName, blobName)`

and write to the archiver stream using `append` ?

Reply:
The callback refers to the saving on the remote storage ([docs](https://azure.github.io/azure-storage-node/BlobService.html#createWriteStreamToBlockBlob)).

We use `append` to add content as you'd expect, but the final result of the operation is given in the callback

Reply:
that's right but the callback is optional (it will write on the remote storage even if you don't provide any). why do we need it ?

Reply:
because otherwise there would not be not a way to know if and when the blob has been saved to the storage. 
The flow is: _zipStream_ --> _blobStream_ --> _blobSaving_. We can listen for the `end` and `finish` event, but still the saving step can fail imho